internal/discord/commands.go

```diff
@@ -0,0 +1,228 @@
+package discord
+
+import (
+	"fmt"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// Kick attempts to kick the user specified user from the server the command was invoked in.
+// Fields:
+//
+//	user: 	User
+//	reason: string
+func (d *Discord) Kick(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	options := i.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options))
+	for _, opt := range options {
+		optionMap[opt.Name] = opt
+	}
+
+	// Log usage of command
+	err := d.CommandLogger(i.Interaction)
+	if err != nil {
+		fmt.Printf("Failed to log: %v\n", err)
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
+
+	// Check if user exists in guild
+	err = d.CheckUserInGuild(i.GuildID, userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = StartInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+
+		return
+	}
+
+	// DM the user regarding the kick
+	channel, err := s.UserChannelCreate(userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not send a DM to user %v", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = StartInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+	} else {
+		// Get guild name
+		var guildname string
+		guild, err := d.Session.Guild(i.GuildID)
+		if err != nil {
+			fmt.Printf("Unable to find guild name: %v", err)
+			guildname = i.GuildID
+		} else {
+			guildname = guild.Name
+		}
+
+		tempstr := fmt.Sprintf("You are being kicked from %v for the reason:\n%v",
+			guildname,
+			optionMap["reason"].StringValue(),
+		)
+
+		_, err = s.ChannelMessageSend(channel.ID, tempstr)
+		if err != nil {
+			tempstr := fmt.Sprintf("Could not send a DM to user %v", userToKick)
+			fmt.Printf("%v: %v\n", tempstr, err)
+
+			err = StartInteraction(s, i.Interaction, tempstr)
+			if err != nil {
+				fmt.Printf("Unable to send ephemeral message: %v\n", err)
+			}
+		}
+	}
+
+	// Attempt to kick user
+	if len(optionMap["reason"].StringValue()) > 0 {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		err = StartInteraction(s, i.Interaction, "Please provide a reason for the kick.")
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+
+		return
+	}
+
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = StartInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+	} else {
```

Comment:
i think it makes more sense to move this up in the func. how i picture it is:
1. user uses the command and fills out the fields
2. we dm the user
3. we kick them from the server

Reply:
That's the same order unless I'm mistaken.
With the changes below, it should be something along these lines:
1. User uses command
2. Bot logs usage of command 
3. Bot attempts to DM user
4. Bot kicks user regardless 
5. If kicked user was not able to be DMed, inform invoker and edit log message to reflect it

Reply:
ah i got error output confused with actual input text. yeah this looks good to me